course/src/main/java/org/openedx/course/presentation/info/CourseInfoFragment.kt

```diff
@@ -0,0 +1,318 @@
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.compose.ui.zIndex
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.openedx.core.UIMessage
+import org.openedx.core.presentation.catalog.CatalogWebViewScreen
+import org.openedx.core.presentation.catalog.WebViewLink.Authority.COURSE_INFO
+import org.openedx.core.presentation.dialog.alert.ActionDialogFragment
+import org.openedx.core.presentation.dialog.alert.InfoDialogFragment
+import org.openedx.core.system.AppCookieManager
+import org.openedx.core.system.connection.NetworkConnection
+import org.openedx.core.ui.BackBtn
+import org.openedx.core.ui.ConnectionErrorView
+import org.openedx.core.ui.HandleUIMessage
+import org.openedx.core.ui.WindowSize
+import org.openedx.core.ui.displayCutoutForLandscape
+import org.openedx.core.ui.rememberWindowSize
+import org.openedx.core.ui.statusBarsInset
+import org.openedx.core.ui.theme.OpenEdXTheme
+import org.openedx.core.ui.theme.appColors
+import org.openedx.core.ui.theme.appTypography
+import org.openedx.core.ui.windowSizeValue
+import org.openedx.course.R
+import org.openedx.course.presentation.CourseRouter
+import org.openedx.core.R as CoreR
+
+class CourseInfoFragment : Fragment() {
+
+    private val viewModel by viewModel<CourseInfoViewModel>()
+    private val edxCookieManager by inject<AppCookieManager>()
+    private val networkConnection by inject<NetworkConnection>()
+    private val router: CourseRouter by inject()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ) = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+            OpenEdXTheme {
+                val uiMessage by viewModel.uiMessage.observeAsState()
```

Comment:
we added a convention to use Flows instead of LiveData for the new development, https://github.com/openedx/openedx-app-android/pull/92, also application became freeze, 
I think we should ideally use one hot subscription of each type (StateFlow, SharedFlow) and use a sealed classes to observe them. @volodymyr-chekyrta

Reply:
Thanks for the reminder. 👍🏻 ✨